File: v3/python/newscatcherapi_client/apis/tags/latest_headlines_api.py
# coding: utf-8

from newscatcherapi_client.apis.tags.latest_headlines_api_generated import LatestHeadlinesApiGenerated
from newscatcherapi_client.search_all_helper import search_all

class LatestHeadlinesApi(LatestHeadlinesApiGenerated):
    
    def get_all(self, **kwargs):
        return search_all(self.get, **kwargs)

    def post_all(self, **kwargs):
        return search_all(self.post, **kwargs)

File: v3/typescript/api/latest-headlines-api.ts
import { LatestHeadlinesApiGenerated, LatestHeadlinesApiGetRequest, LatestHeadlinesApiPostRequest } from "./latest-headlines-api-generated";
import { searchAll } from "../search-all-helper";
import { AxiosRequestConfig } from "axios";
export * from "./latest-headlines-api-generated";

export class LatestHeadlinesApi extends LatestHeadlinesApiGenerated {

    public async get_all(requestParameters: LatestHeadlinesApiGetRequest, options?: AxiosRequestConfig) {
        return searchAll(this.get.bind(this), requestParameters, options);
    }

    public async post_all(requestParameters: LatestHeadlinesApiPostRequest, options?: AxiosRequestConfig) {
        return searchAll(this.post.bind(this), requestParameters, options);
    }
}
